Compiler.Lexer/Compiler.Lexer/Tokenizer.cs

```diff
@@ -0,0 +1,403 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Compiler.Lexer
+{
+    public partial class Tokenizer
+    {
+        private CharStream Stream;
+        private Token LastToken;
+
+        public Tokenizer(string source)
+        {
+            Stream = new CharStream(source);
+            LastToken = new Token
+                (TokenKind.UnknownToken, string.Empty, 0, 0);
+        }
+
+        public Token Get()
+        {
+            Token get;
+            int lineNumber = Stream.CurrentLineNumber;
+            int position = Stream.CurrentPosition;
+
+            do
+            {
+                if (Stream.EOF)
+                {
+                    return null;
+                }
+            } while (IsWhiteSpace());
+
+            get = IsSymbol();
+            if (get != null)
+            {
+                return get;
+            }
+
+            get = IsOperator();
+            if (get != null)
+            {
+                return get;
+            }
+
+            get = IsIdentifier();
+            if (get != null)
+            {
+                return get;
+            }
+
+            get = IsCharacterLiteral();
+            if (get != null)
+            {
+                return get;
+            }
+
+            get = IsStringLiteral();
+            if (get != null)
+            {
+                return get;
+            }
+
+            get = IsNumericLiteral();
+            if (get != null)
+            {
+                return get;
+            }
+
+            get = IsAddressLiteral();
+            if (get != null)
+            {
+                return get;
+            }
+
+            get = new Token(TokenKind.UnknownToken, Stream.Get().ToString(),
+                lineNumber, position);
+            return get;
+
+        }
+
+        public Token Peek()
+        {
+            int line = Stream.CurrentLineNumber;
+            int position = Stream.CurrentPosition;
+            Token previousToken = LastToken;
+            Token peek = Get();
+            Stream.CurrentLineNumber = line;
+            Stream.CurrentPosition = position;
+            LastToken = previousToken;
+            return peek;
+        }
+
+        private bool IsWhiteSpace()
+        {
+            bool isWhiteSpace = false;
+            while (Stream.Peek() != null && char.IsWhiteSpace((char)Stream.Peek()))
+            {
+                isWhiteSpace = true;
+                Stream.Get();
+            }
+            return isWhiteSpace;
+        }
+
+        private Token IsSymbol()
+        {
```

Comment:
method name of Is suggests a Boolean return type. Maybe rename it to GetSymbol, and others such as IsOperator etc